pnc-web/src/main/webapp/WEB-INF/web.xml

```diff
@@ -1,6 +1,32 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
-         version="3.1">
+<web-app xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://xmlns.jcp.org/xml/ns/javaee" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd" version="3.1">
+  <context-param>
+    <param-name>resteasy.role.based.security</param-name>
+    <param-value>true</param-value>
+  </context-param>
+  <listener>
+    <listener-class>org.jboss.resteasy.plugins.server.servlet.ResteasyBootstrap</listener-class>
+  </listener>
+  <security-constraint>
+    <web-resource-collection>
+      <web-resource-name>pnc-servlet</web-resource-name>
+      <url-pattern>/*</url-pattern>
+      <http-method>GET</http-method>
+      <http-method>POST</http-method>
+    </web-resource-collection>
+    <auth-constraint>
+      <role-name>admin</role-name>
```

Comment:
Hey Sebastian,

well I am learning as well ;)
However, this is just a start... EAP security domains can be configured 
for either Kerberos or LDAP or for any desired Authentication Provider 
we will choose in the future.
OAuth for Rest is on of the possibilities, it depends on what the 
requirements are
Rest endpoints can be also secured via @RoleAllowed annotations, once we 
define the
roles of endpoints

I don't know much in detail about Keycloak, but I will have a look into it.

SSO, would come in the future if there is need for it, once we do have 
more web sites ?
In case there is plan to... currently we have pnc-web only.

I suggest go step-by-step and not introduce a pollute NCL with things, 
we do not necessarily be using now.

I am going to dig deeper and elaborate on this

One more thing... we need to split this into more issues, and well 
define, what and when to implement.
P.

On 11.3.2015 21:23, Sebastian Łaskawiec wrote:
>
> In pnc-web/src/main/webapp/WEB-INF/web.xml 
> <https://github.com/project-ncl/pnc/pull/215#discussion_r26251417>:
>
> > +  <context-param>
> > +    <param-name>resteasy.role.based.security</param-name>
> > +    <param-value>true</param-value>
> > +  </context-param>
> > +  <listener>
> > +    <listener-class>org.jboss.resteasy.plugins.server.servlet.ResteasyBootstrap</listener-class>
> > +  </listener>
> > +  <security-constraint>
> > +    <web-resource-collection>
> > +      <web-resource-name>pnc-servlet</web-resource-name>
> > +      <url-pattern>/*</url-pattern>
> > +      <http-method>GET</http-method>
> > +      <http-method>POST</http-method>
> > +    </web-resource-collection>
> > +    <auth-constraint>
> > +      <role-name>admin</role-name>
>
> Hey! I'm definitely not an expert in case of 
> authentication/authorization but shouldn't we have a Keycloak server 
> in our environment? We will probably need LDAP integration, OAuth for 
> REST and probably Kerberos integration or SSO integration. Seems to me 
> like a perfect scenario for Keycloak.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/project-ncl/pnc/pull/215/files#r26251417>.
>